iceoryx_dust/include/iceoryx_dust/cxx/std_string_support.hpp

```diff
@@ -19,10 +19,38 @@
 #include "iox/into.hpp"
 #include "iox/string.hpp"
 
+#include <ostream>
 #include <string>
 
 namespace iox
 {
+
+template <>
+struct is_custom_string<std::string> : public std::true_type
+{
+};
+
+namespace internal
+{
+template <>
+struct GetData<std::string>
+{
+    static const char* call(const std::string& data) noexcept
+    {
+        return data.data();
+    }
+};
+
+template <>
+struct GetSize<std::string>
+{
+    static uint64_t call(const std::string& data) noexcept
+    {
+        return data.size();
+    }
+};
```

Comment:
It seems three is currently no default implementation for `GetSize` and `GetData`. What do you think about adding a default implementation with a `static_assert(always_false_v<T>, "Please implement bla bla");`?

I'm not sure if something similar could be done for `is_custom_string` in order to have a sane error message on compilations.